linchpin/api/__init__.py

```diff
@@ -11,111 +17,165 @@ class LinchpinError(Exception):
+
+
+    def get_evar(self, key=None):
+        """
+        Get the current evars (extra_vars)
+
+        :param key: key to use
+        """
+
+        if key:
+            return self.ctx.evars.get(key, None)
+        return self.evars
 
 
-    def run_playbook(self, pinfile, targets='all', playbook='provision'):
+    def set_evar(self, key, value):
+        """
+        Set a value into evars (extra_vars). Does not persist into a file,
+        only during the current execution.
+
+        :param key: key to use
+
+        :param value: value to set into evars
+        """
+
+        self.set_cfg('evars', key, value)
+        self.evars[key] = value
+
+
+    def run_playbook(self, pinfile, targets='all', playbook='up'):
 
 
         """
         This function takes a list of targets, and executes the given
         playbook (provison, destroy, etc.) for each provided target.
 
-        \b
-        pf:
-            Provided PinFile, with available targets,
+        :param pinfile: Provided PinFile, with available targets,
 
-        \b
-        targets:
-            A tuple of targets to run.
+        :param targets: A tuple of targets to run. (default: 'all')
         """
 
         pf = yaml2json(pinfile)
 
-        # playbooks check whether from_cli is defined
+        # playbooks check whether from_api is defined
         # if not, vars get loaded from linchpin.conf
-        self.ctx.evars['from_cli'] = True
+
+
+        self.ctx.log_debug('from_api: {0}'.format(self.get_evar('from_api')))
+
         self.ctx.evars['lp_path'] = self.lp_path
 
+        self.console = eval(self.ctx.cfgs['ansible'].get('console', 'False'))
+
+        if not self.console:
+            self.console = self.ctx.verbose
+
+        print('console: {0}'.format(self.console))
+
         self.ctx.evars['default_resources_path'] = '{0}/{1}'.format(
                                 self.ctx.workspace,
-                                self.ctx.cfgs['evars']['resources_folder'])
+                                self.ctx.evars['resources_folder'])
         self.ctx.evars['default_inventories_path'] = '{0}/{1}'.format(
                                 self.ctx.workspace,
-                                self.ctx.cfgs['evars']['inventories_folder'])
+                                self.ctx.evars['inventories_folder'])
 
         self.ctx.evars['state'] = "present"
 
         if playbook == 'destroy':
             self.ctx.evars['state'] = "absent"
 
 
-        # checks wether the targets are valid or not
+        # checks whether the targets are valid or not
         if set(targets) == set(pf.keys()).intersection(targets) and len(targets) > 0:
             for target in targets:
-                self.ctx.log_state('{0} target: {1}'.format(playbook, target))
+                self.ctx.log_state('target: {0}, action: {1}'.format(target, playbook))
                 topology = pf[target]['topology']
                 topology_registry = pf.get("topology_registry", None)
```

Comment:
@herlo : topology_registry _ variable is not used anywhere ! am i missing something ?